pkg/virt-controller/watch/vm.go

```diff
@@ -3077,6 +3204,21 @@ func (c *VMController) handleMemoryHotplugRequest(vm *virtv1.VirtualMachine, vmi
 		return fmt.Errorf("memory hotplug is not allowed while VMI is migrating")
 	}
 
+	// If the following is true, MaxGuest was calculated, not manually specified (or the validation webhook would have rejected the change).
+	// Since we're here, we can also assume MaxGuest was not changed in the VM spec since last boot.
+	// Therefore, bumping Guest to a value higher than MaxGuest is fine, it just requires a reboot.
+	if vm.Spec.Template.Spec.Domain.Memory.Guest != nil && vmi.Spec.Domain.Memory.MaxGuest != nil &&
+		vm.Spec.Template.Spec.Domain.Memory.Guest.Cmp(*vmi.Spec.Domain.Memory.MaxGuest) == 1 {
+		vmConditions := controller.NewVirtualMachineConditionManager()
+		vmConditions.UpdateCondition(vm, &virtv1.VirtualMachineCondition{
+			Type:               virtv1.VirtualMachineRestartRequired,
+			LastTransitionTime: v1.Now(),
+			Status:             k8score.ConditionTrue,
+			Message:            "memory updated in template spec to a value higher than what's available",
+		})
+		return nil
```

Comment:
Do we miss a check if the guest memory is lower than the initial boot memory?

Reply:
Yes, but I think we've decided not to support the unplug case .
@acardace do you remember?